src/adapters/ActivitiesJSONAdapter.js

```diff
@@ -59,4 +69,40 @@ export default class ActivitiesJSONAdapter extends ActivitiesAdapter {
       observer.complete();
     });
   }
+
+  /**
+   * Posts an attachment action, returns an observable that emits the created action
+   *
+   * @param {string} activityID  ID of the activity corresponding to this submit action
+   * @param {object} inputs  The message content
+   * @returns {Observable.<object>} Observable stream that emits data of the newly created action
+   */
+  postAction(activityID, inputs) {
+    return Observable.create((observer) => {
+      const activity = this.datasource[activityID];
+
+      if (activity) {
+        const actionID = `action-${activityID}-${this.datasource[activityID]?.actions.length + 1}`;
```

Comment:
you already have this.datasource[activityID] in a variable `activity`

same for extracting roomID below